examples/appSpec/zomes/three/capabilities/main/code/index.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  serialize,
+  deserialize,
+  debug,
+  commit_entry,
+  get_entry,
+  check_encoded_allocation
+} from "../../../../../../../index"
+//TODO: Remove this relative import and link to node_modules. Ok for dev
+
+
+
+/*----------  Public Functions  ----------*/
+
+
+export function test_debug(encoded_allocation: u32): u32 {
+  // necessary for setting memory module to at least one page
+  // find another way
+  const tree = "test";
+  let val: string = deserialize(encoded_allocation);
+  debug(val);
+  return 0;
+}
+
+export function test_commit(encoded_allocation: u32): u32 {
+  let val: string = deserialize(encoded_allocation);
+  // TODO: how do we parse JSON input here, to get custom values?
```

Comment:
@willemolding this one of the biggest looming questions for me, how do we parse a single string that represents a json value into its multiple values. we might need to write our own little assemblyscript compatible parser